tests/run-make/rustdoc-verify-output-files/rmake.rs

```diff
@@ -0,0 +1,67 @@
+use std::fs::{copy, read, read_dir};
+use std::path::{Path, PathBuf};
+
+use run_make_support::{copy_dir_all, rustdoc, tmp_dir};
+
+fn generate_docs(out_dir: &Path, json_output: bool) {
+    let mut rustdoc = rustdoc();
+    rustdoc.input("src/lib.rs").crate_name("foobar").crate_type("lib").out_dir(&out_dir);
+    if json_output {
+        rustdoc.arg("-Zunstable-options").output_format("json");
+    }
+    rustdoc.run();
+}
+
+fn read_file(path: PathBuf) -> Vec<u8> {
+    match read(&path) {
+        Ok(c) => c,
+        Err(e) => panic!("Failed to read `{}`: {:?}", path.display(), e),
+    }
+}
+
+fn recursive_diff(dir1: &Path, dir2: &Path) {
+    for entry in read_dir(dir1).unwrap() {
+        let entry = entry.unwrap();
+        let entry_name = entry.file_name();
+        let path = entry.path();
+
+        if path.is_dir() {
+            recursive_diff(&path, &dir2.join(entry_name));
+        } else {
+            let file1 = read_file(path);
+            let file2 = read_file(dir2.join(entry_name));
+
+            assert!(file1 == file2);
```

Comment:
Suggestion: could we add a bit of context about what failed, i.e.

```rs
assert!(file1 == file2, "`{path}` and `{path2}` not equal");
```

Reply:
Why not use `assert_eq!` macro so that it will explained by default when it fails?

Reply:
Because there are minified files, so that wouldn't be a useful information. However, displaying paths is a very good idea.